otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py

```diff
@@ -0,0 +1,331 @@
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.polygon import Polygon
+from otx.api.entities.subset import Subset
+from otx.api.utils.shape_factory import ShapeFactory
+from pytorch_lightning import LightningDataModule
+
+logger = get_logger()
+
+
+class OTXVIsualPromptingDataset(Dataset):
+    """Visual Prompting Dataset Adaptor.
+
+    Args:
+        dataset (DatasetEntity): Dataset entity.
+        transform (MultipleInputsCompose): Transformations to apply to the dataset.
+        offset_bbox (int): Offset to apply to the bounding box, defaults to 0.
+    """
+
+    def __init__(self, dataset: DatasetEntity, transform: MultipleInputsCompose, offset_bbox: int = 0) -> None:
+
+        self.dataset = dataset
+        self.transform = transform
+        self.offset_bbox = offset_bbox
+
+        self.labels = dataset.get_labels()
+
+    def __len__(self) -> int:
+        """Get size of the dataset.
+
+        Returns:
+            int: Size of the dataset.
+        """
+        return len(self.dataset)
+
+    def __getitem__(self, index: int) -> Dict[str, Union[int, List, Tensor]]:
+        """Get dataset item.
+
+        Args:
+            index (int): Index of the dataset sample.
+
+        Returns:
+            Dict[str, Union[int, List, Tensor]]: Dataset item.
+        """
+        dataset_item = self.dataset[index]
+        item: Dict[str, Union[int, Tensor]] = {"index": index}
+
+        width, height = dataset_item.width, dataset_item.height
+        bboxes: List[List[int]] = []
+        points: List = []  # TBD
+        gt_masks: List[np.ndarray] = []
+        labels: List[ScoredLabel] = []
+        for annotation in dataset_item.get_annotations(labels=self.labels, include_empty=False, preserve_id=True):
+            if isinstance(annotation.shape, Image):
+                # use mask as-is
+                gt_mask = annotation.shape.numpy.astype(np.uint8)
+            elif isinstance(annotation.shape, Polygon):
+                # convert polygon to mask
+                gt_mask = self.convert_polygon_to_mask(annotation.shape, width, height)
+            else:
+                continue
+
+            if gt_mask.sum() == 0:
+                # pass no gt
+                continue
+            gt_masks.append(gt_mask)
+
+            # generate bbox based on gt_mask
+            bbox = self.generate_bbox_from_mask(gt_mask, width, height)
+            bboxes.append(bbox)
+
+            # TODO (sungchul): generate random points from gt_mask
+
+            # add labels
+            labels.extend(annotation.get_labels(include_empty=False))
+
+        if len(gt_masks) == 0:
+            return {
+                "images": [],
+                "bboxes": [],
+                "points": [],
+                "gt_masks": [],
+                "original_size": [],
+                "path": [],
+                "labels": [],
+            }
+
+        bboxes = np.array(bboxes)
+        item.update(
```

Comment:
how about reusing dataset_item instead of creating item from scratch? The current implementation is expected to have overhead when reading data `images=dataset_item.numpy`. I think updating only gt_masks, bboxes,points, and labels is enough for `__getitem__`

Reply:
`dataset_item` is `DatasetItemEntity` and it has its own instances and behaves differently than `dict`.